sdk/python/tests/README.md

```diff
@@ -0,0 +1,211 @@
+* See the [custom offline store demo](https://github.com/feast-dev/feast-custom-offline-store-demo) and the [custom online store demo](https://github.com/feast-dev/feast-custom-online-store-demo) for examples.
+
+### To include a new offline / online store in the main Feast repo
+
+* Extend `data_source_creator.py` for your offline store.
+* In `repo_configuration.py` add a new`IntegrationTestRepoConfig` or two (depending on how many online stores you want to test).
+* Run the full test suite with `make test-python-integration.`
+
+### Including a new offline / online store in the main Feast repo from external plugins with community maintainers.
+
+* This folder is for plugins that are officially maintained with community owners. Place the APIs in feast/infra/offline_stores/contrib/.
+* Extend `data_source_creator.py` for your offline store and implement the required APIs.
+* In `contrib_repo_configuration.py` add a new `IntegrationTestRepoConfig` (depending on how many online stores you want to test).
+* Run the test suite on the contrib test suite with `make test-python-contrib-universal`.
+
+### To include a new online store
+
+* In `repo_configuration.py` add a new config that maps to a serialized version of configuration you need in `feature_store.yaml` to setup the online store.
+* In `repo_configuration.py`, add new`IntegrationTestRepoConfig` for offline stores you want to test.
+* Run the full test suite with `make test-python-integration`
+
+### To use custom data in a new test
+
+* Check `test_universal_types.py` for an example of how to do this.
+
+```python
+@pytest.mark.integration
+def your_test(environment: Environment):
+    df = #...#
+    data_source = environment.data_source_creator.create_data_source(
+        df,
+        destination_name=environment.feature_store.project
+    )
+    your_fv = driver_feature_view(data_source)
+    entity = driver(value_type=ValueType.UNKNOWN)
+    fs.apply([fv, entity])
+
+    # ... run test
+```
+
+### Running your own redis cluster for testing
+
+* Install redis on your computer. If you are a mac user, you should be able to `brew install redis`.
+    * Running `redis-server --help` and `redis-cli --help` should show corresponding help menus.
+* Run `cd scripts/create-cluster` and run `./create-cluster start` then `./create-cluster create` to start the server. You should see output that looks like this:
+~~~~
+Starting 6001
+Starting 6002
+Starting 6003
+Starting 6004
+Starting 6005
+Starting 6006
+~~~~
+* You should be able to run the integration tests and have the redis cluster tests pass.
+* If you would like to run your own redis cluster, you can run the above commands with your own specified ports and connect to the newly configured cluster.
+* To stop the cluster, run `./create-cluster stop` and then `./create-cluster clean`.
```

Comment:
nit but this is a strange command (`stop`) to pass into a script called `create-cluster`. Maybe we should call it `redis-cluster` or something?

Reply:
Sure let me change